holidays/generate_ics.py

```
@@ -0,0 +1,170 @@
+        self.entity_loader = entity_loader
+
+    def validate_years(self) -> None:
+        start_year, end_year = self.entity_loader().start_year, self.entity_loader().end_year
+        min_year, max_year = min(self.args.years), max(self.args.years)
+        if min_year < start_year or max_year > end_year:
+            year_part = (
+                f"year {min_year} is not supported"
+                if min_year == max_year
+                else f"year range {min_year}-{max_year} is not fully supported"
+            )
+            warnings.warn(
```

Ruff (0.15.15)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)